src/routes/safe/store/actions/utils.js

```diff
@@ -0,0 +1,51 @@
+// @flow
+import axios from 'axios'
+
+import { buildTxServiceUrl } from '~/logic/safe/transactions/txHistory'
+import { getGnosisSafeInstanceAt } from '~/logic/contracts/safeContracts'
+
+export const getLastTx = async (safeAddress: string): Promise<TransactionProps> => {
+  try {
+    const url = buildTxServiceUrl(safeAddress)
+    const response = await axios.get(url, { params: { limit: 1 } })
+
+    return response.data.results[0]
+  } catch (e) {
+    console.error('failed to retrieve last Tx from server', e)
+    return null
+  }
+}
+
+export const getSafeNonce = async (safeAddress: string): Promise<string> => {
+  // use current's safe nonce as fallback
+  const safeInstance = await getGnosisSafeInstanceAt(safeAddress)
+  return (await safeInstance.nonce()).toString()
+}
+
+export const getNewTxNonce = async (txNonce, lastTx, safeAddress) => {
+  if (!Number.isInteger(Number.parseInt(txNonce, 10))) {
+    return lastTx === null ? getSafeNonce(safeAddress) : `${lastTx.nonce + 1}`
+  }
+  return txNonce
+}
+
+export const doesTxNeedApproval = async (safeInstance, nonce, lastTx) => {
```

Comment:
also, changed this name and added some comments to the conditions

Reply:
:1st_place_medal:  better name than the one we come out with yesterday